compiler/jsgen.nim

```diff
@@ -1829,7 +1829,12 @@ proc createObjInitList(p: PProc, typ: PType, excludedFieldIDs: IntSet, output: v
 proc arrayTypeForElemType(conf: ConfigRef; typ: PType): string =
   let typ = typ.skipTypes(abstractRange)
   case typ.kind
-  of tyInt, tyInt32: "Int32Array"
+  of tyInt:
+    if $typ.sym.loc.r == "bigint":
```

Comment:
Preexisting from #16606, see https://github.com/nim-lang/Nim/blob/3294758da4f98179a83ab004cc3e5c65483af55d/compiler/jsgen.nim#L1869-L1872

Maybe add a magic?

Reply:
Ok so we either

1. merge this and suffer the consequences to fix an issue no one has even reported beyond a code comment in a PR that wanted bigints for a purpose they're no longer needed for and still found a workaround
2. leave it as is (initializer still checks for `importjs: "bigint"`), people can use `array[JsRoot]` with `cast` as a workaround
3. remove the `importjs: "bigint"` thing and replace it with something that inits as `null` and either is `requiredInit` or `not nil` (not sure if either are even reliable) or just gives bad JS errors when attempting to use a null value, this also breaks code
4. leave it as is (just the initializer) and introduce a mechanism for array types to not use number arrays and force people to use it every time they want an array of `JsBigInt` (lest it errors at runtime)
5. introduce a mechanism for array element types to opt out of number arrays (???)
6. introduce a mechanism for types to include initializer codegen (?????)
7. specialize for `JsBigInt` in a more streamlined way in the compiler for which one needs to think of a good, cheap and silent way to do
8. remove the `importjs: "bigint"` thing and make the `jsbigint` library add runtime workarounds for the default values which aren't necessarily always applicable

I'm just going to leave it alone because I never want to think about this again

Reply:
> I'm just going to leave it alone because I never want to think about this again

Welcome to my world. ;-)